queue/remote_base.go

```diff
@@ -248,6 +249,46 @@ func (q *remoteBase) Complete(ctx context.Context, j amboy.Job) {
 	}
 }
 
+func (q *remoteBase) CompleteRetry(ctx context.Context, j amboy.RetryableJob) error {
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	const retryInterval = time.Second
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+
+	var attempt int
+	catcher := grip.NewBasicCatcher()
+
+	for attempt = 1; attempt <= 10; attempt++ {
+		select {
+		case <-ctx.Done():
+			catcher.Add(ctx.Err())
+			return errors.Wrapf(catcher.Resolve(), "giving up after attempt %d", attempt)
+		case <-timer.C:
+			j.UpdateRetryInfo(amboy.JobRetryOptions{
+				NeedsRetry: utility.FalsePtr(),
+			})
+
+			if err := q.driver.Complete(ctx, j); err != nil {
+				catcher.Wrapf(err, "attempt %d", attempt)
+				if isMongoDupKey(err) {
+					j.AddError(catcher.Resolve())
+					return errors.Wrapf(catcher.Resolve(), "giving up after attempt %d", attempt)
```

Comment:
Nit: you can just break here and let the `return` on line 289 take care of this.

Reply:
Same thing as above.